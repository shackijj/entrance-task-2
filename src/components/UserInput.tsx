import * as React from 'react';
import User, { UserProps } from './User';
import { Close } from './GlyphIcon/GlyphIcon';
import './UserInput.css';

interface UserInputProps extends UserProps {
  onCloseClick?: () => void;
}

const UserInput: React.SFC<UserInputProps> = ({avatarUrl, login, onCloseClick}) => (
  <div className="UserInput">
    <User login={login} avatarUrl={avatarUrl}/>
    <Close classes={['UserInput-Close']} onClick={onCloseClick ? onCloseClick : undefined}/>
  </div>
);

export default UserInput;