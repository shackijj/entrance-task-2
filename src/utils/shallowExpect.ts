import { compose } from 'redux';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

export default compose(expect, toJSON, shallow);