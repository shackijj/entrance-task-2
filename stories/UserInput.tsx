import * as React from 'react';
import { storiesOf } from '@storybook/react';
import UserInput from '../src/components/UserInput';

storiesOf('UserInput', module)
  .add('default', () => (
  <UserInput
    name="Артур Пирожков"
    avatarUrl="https://pp.userapi.com/c402317/v402317531/6107/Tg1PWZHDAO0.jpg"
  />));