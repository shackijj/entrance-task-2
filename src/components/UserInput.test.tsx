import * as React from 'react';
import UserInput from './UserInput';
import shallowExpect from '../utils/shallowExpect';
import { mount } from 'enzyme';

describe('UserInput', () => {
  it('should match snapshpt', () => {
    shallowExpect(<UserInput login="asd" avatarUrl="qwe"/>).toMatchSnapshot();
  });
  it('should fire onCloseClick', () => {
    const mock = jest.fn();
    const wrapper = mount(<UserInput login="asd" avatarUrl="qwe" onCloseClick={mock}/>);
    expect(mock).toHaveBeenCalledTimes(0);
    wrapper.find('.UserInput-Close').simulate('click');
    expect(mock).toHaveBeenCalledTimes(1);
  });
});