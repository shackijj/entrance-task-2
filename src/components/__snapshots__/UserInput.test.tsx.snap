// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInput should match snapshpt 1`] = `
<div
  className="UserInput"
>
  <User
    avatarUrl="qwe"
    login="asd"
  />
  <Component
    classes={
      Array [
        "UserInput-Close",
      ]
    }
  />
</div>
`;
